### PR TITLE
Refactor LoanTime to extend DataField class

### PR DESCRIPTION
**Pull request by xantho09**
Fix issue #76

### CODE REVIEW

**Comment by wn96** [comment]
Please use the issue flow. Create an issue and refer to it.

File: src/main/java/seedu/address/model/loan/LoanTime.java [hunk 1]
```diff
@@ -35,133 +32,203 @@
...
      * Function returns 0 if the specified other LoanTime is before the current LoanTime.
      *
      * @param otherTime LoanTime object to be compared to.
      */
     public long loanTimeDifferenceMinutes(LoanTime otherTime) {
-        long timeDifference = (otherTime.value.toEpochMilli() - this.value.toEpochMilli());
-        return (timeDifference >= 0) ? timeDifference / 60000 : 0;
+        long timeDifference = Duration.between(this.value, otherTime.value).toMinutes();
+        return (timeDifference >= 0) ? timeDifference : 0;
     }
```
Comment (FongYuan) [inline src/main/java/seedu/address/model/loan/LoanTime.java]: This could re-use the static method following it, to reduce repeated code.

Comment (xantho09) [reply src/main/java/seedu/address/model/loan/LoanTime.java]: That's true. I'll make that change.

File: src/main/java/seedu/address/model/loan/LoanTime.java [hunk 2]
```diff
@@ -35,133 +32,202 @@
...
+        // Consider both cases.
+        if (isValidLongLoanTimeFormat(objString)) {
+            // Split by spaces to separate the Date and Time strings
+            String[] dateData = objString.split(" ");
+
+            assert dateData.length == 2;
+
+            // Obtain the Date and Time strings
+            String loanDateString = dateData[0];
+            String loanTimeString = dateData[1];
```
Comment (FongYuan) [inline src/main/java/seedu/address/model/loan/LoanTime.java]: I noticed that this is the same process as line 83. Perhaps some abstraction/method extraction is appropriate.

**Pull request merged**